sa/database_test.go

```diff
@@ -73,3 +73,26 @@ func TestStrictness(t *testing.T) {
 		t.Fatalf("Got wrong type of error: %s", err)
 	}
 }
+
+func TestSafeSelectOne(t *testing.T) {
```

Comment:
Actually thinking about this it'd be super easy with just the method so I'm going to do that!